### PR TITLE
Consolidate job clone API tests

### PR DESCRIPTION
**Pull request by kalikiana**
- The Tests in `09-job_clone` are really API tests
- `api/04-jobs` is getting big enough to split up
- `api/04-jobs-clone` can host cloning-related unit tests
- Move group_overview cases to `10-jobgroups.t`

### CODE REVIEW

File: t/09-job_clone.t [hunk 1]
```diff
@@ -1,61 +0,0 @@
...
-is($minimalx->duplicate, undef, "cannot clone after reloading");
-
-# But cloning the clone should be possible after job state change
-$clone->state(OpenQA::Jobs::Constants::CANCELLED);
-$clones = $clone->duplicate({prio => 35});
-my $second = $rset->find($clones->{$clone->id}->{clone});
-is($second->TEST,     "minimalx", "same test again");
-is($second->priority, 35,         "with adjusted priority");
-
-done_testing();
```
Comment (okurz) [inline t/09-job_clone.t:61]: I think you might have misunderstood what is meant with "t/api". This test module was correct in the way that it does not use the actual web API at all so maybe the decision to move it to t/api/04-jobs-clone.t was not correct after all.

File: t/api/04-jobs-clone.t [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+is($minimalx->duplicate, undef, "cannot clone after reloading");
+
+# But cloning the clone should be possible after job state change
+$clone->state(OpenQA::Jobs::Constants::CANCELLED);
+$clones = $clone->duplicate({prio => 35});
+my $second = $jobs->find($clones->{$clone->id}->{clone});
+is($second->TEST,     "minimalx", "same test again");
+is($second->priority, 35,         "with adjusted priority");
+
+subtest 'handle settings when posting job' => sub {
```
Comment (Amrysliu) [inline t/api/04-jobs-clone.t:71]: I think this test code is not related with `clone` operation. `Handle settings` happens when creating a job using `job post`. And `clone job` does not handle settings, it just uses the settings (copy from old job) to create a new one. I guess you put this test code here because the next one `do not re-generate settings when cloning job` depends on it.

File: t/09-job_clone.t (hunk 1 above)
Comment (kalikiana) [reply t/09-job_clone.t:61]: - It does now that I'm consolidating it.
- It also uses Script/CloneJob - to avoid making the main CloneJob test slower.

My original goal was to address gaps in test coverage of the CloneJob module. Maybe you want to take over and place them where you see fit?

File: t/api/04-jobs-clone.t (hunk 2 above)
Comment (kalikiana) [reply t/api/04-jobs-clone.t:71]: Yeah. They kinda come in tandem.
